libexec/rtld-elf/rtld.c

```diff
@@ -2870,6 +2874,10 @@ parse_rtld_phdr(Obj_Entry *obj)
 	for (ph = obj->phdr;  (const char *)ph < (const char *)obj->phdr +
 	    obj->phsize; ph++) {
 		switch (ph->p_type) {
+		case PT_LOAD:
+			obj->mapsize = rtld_max(obj->mapsize,
```

Comment:
In theory RTLD supports being built as non-PIC (see d5b537d01a6711c4426da2a627b11a71cf3b11e8 from 1999) but I don't think this has ever been done? It's probably bitrotted too in the past 26 years, if it ever worked. But at the same time, it might be nice to set vaddrbase and mapsize the same as is done for normal objects?